packages/mdc-segmented-button/segmented-button/component.ts

```diff
@@ -22,44 +22,122 @@
  */
 
 import {MDCComponent} from '@material/base/component';
-import {MDCSegmentedButtonSegmentFactory} from '../segment/component';
-import {SegmentDetail} from '../types';
+import {CustomEventListener} from '@material/base/types';
+import {MDCSegmentedButtonSegment, MDCSegmentedButtonSegmentFactory} from '../segment/component';
+import {SegmentDetail, MDCSegmentedButtonEvent} from '../types';
+import {MDCSegmentedButtonAdapter} from './adapter';
 import {MDCSegmentedButtonFoundation} from './foundation';
+import {strings} from './constants';
 
 export class MDCSegmentedButton extends MDCComponent<MDCSegmentedButtonFoundation> {
   static attachTo(root: Element): MDCSegmentedButton {
     return new MDCSegmentedButton(root);
   }
 
-  initialize(_segmentFactory: MDCSegmentedButtonSegmentFactory) {
-    return;
+  get segments(): ReadonlyArray<MDCSegmentedButtonSegment> {
+    return this.segments_.slice();
+  }
+
+  private segments_!: MDCSegmentedButtonSegment[]; // assigned in initialize
+  private segmentFactory!:
+      (el: Element) => MDCSegmentedButtonSegment; // assigned in initialize
+  private handleSelected!:
+      CustomEventListener<MDCSegmentedButtonEvent>; // assigned in initialSyncWithDOM
+
+  initialize(segmentFactory: MDCSegmentedButtonSegmentFactory = (el) => new MDCSegmentedButtonSegment(el)) {
+    this.segmentFactory = segmentFactory;
+    this.segments_ = this.instantiateSegments(this.segmentFactory);
+  }
+
+  private instantiateSegments(segmentFactory: MDCSegmentedButtonSegmentFactory): MDCSegmentedButtonSegment[] {
+    const segmentElements: Element[] =
+        [].slice.call(this.root.querySelectorAll(strings.SEGMENT_SELECTOR));
+    return segmentElements.map((el: Element) => segmentFactory(el));
   }
 
   initialSyncWithDOM() {
-    return;
+    this.handleSelected = (event) =>
+        this.foundation.handleSelected(event.detail);
+    this.listen(strings.SELECTED_EVENT, this.handleSelected);
+
+    // foundation is undefined in #instantiateSegments
```

Comment:
Didn't quite understand this code comment.

Reply:
I wanted to include this portion of code in `instantiateSegments` (called in `initialize`) but `this.foundation` appeared to be `undefined`. So I moved it here and left a comment so I wouldn't forget and try to move it back. I'll remove the comment.